fs/fcb/src/fcb_append.c

```diff
@@ -36,7 +57,7 @@ fcb_new_area(struct fcb *fcb, int cnt)
         if (!rfa) {
             rfa = fa;
         }
-        if (fa == fcb->f_oldest) {
+        if (fcb->f_oldest == fcb_area_past_newest(fcb)) {
```

Comment:
@mkiiskila @vrahane Thoughts on handling this? I'm sure not everyone is going to want to check a sector ahead.

Reply:
This change here would break things for FCB user(s) which rely on f_scratch_cnt keeping a scratch sector around (i.e. config).

I believe you're trying to make sure that log rotate notify gets called when there is still one 'scratch' area left empty. You should do that by setting f_scratch_cnt to value 1 or greater before calling fcb_init().

The intent is to create new log entries from data which exists within the flash area about to get erased by a call to fcb_rotate(). The way to achieve would be to:
1) set f_scratch_cnt to 1 before calling fcb_init()
2) in the log_rotate callback call fcb_append_to_scratch() to 'activate' the scratch area as a possible area for writes
3) then you can write log entries to this scratch area.

There's an example of this process in conf_fcb_compress_internal()

Reply:
Agreed. I went with your suggestion.